File: orders-private/orders-cancel/orders-cancel/orders-cancel.py
import os
import requests
import hmac
import hashlib
import base64
import json
import time
from dotenv import load_dotenv

load_dotenv()

def cancel_orders(orders):
    api_key = os.getenv('API_KEY')
    secret_key = os.getenv('API_SECRET').encode('utf-8')
    ts = time.strftime('%Y-%m-%dT%H:%M:%S', time.gmtime())
    nonce = str(int(time.time() * 1000))
    method = "/v3/orders/cancel"
    api_url = "api.ox.fun"

    post_data = {
        "recvWindow": 200000,
        "responseType": "FULL",
        "timestamp": int(time.time() * 1000),
        "orders": orders
    }

    body = json.dumps(post_data)
    msg_string = f"{ts}\n{nonce}\nDELETE\n{api_url}\n{method}\n{body}"

    sign = base64.b64encode(hmac.new(secret_key, msg_string.encode('utf-8'), hashlib.sha256).digest()).decode('utf-8')

    headers = {
        'Content-Type': 'application/json',
        'AccessKey': api_key,
        'Timestamp': ts,
        'Signature': sign,
        'Nonce': nonce
    }

    try:
        response = requests.delete(f"https://{api_url}{method}", data=body, headers=headers)
        response.raise_for_status()
        print('Response:', response.json())
    except requests.exceptions.RequestException as error:
        print('Error canceling orders:', error)

# Example usage
orders_to_cancel = [
    {
        "marketCode": "BTC-USD-SWAP-LIN",
        "orderId": "1111111111111",
        "clientOrderId": 11111111111111
    }
]

cancel_orders(orders_to_cancel)

File: orders-private/orders-cancel/orders-cancel/orders-cancel.js
require('dotenv').config();
const axios = require('axios');
const crypto = require('crypto');

async function cancelOrders(orders) {
  const apiKey = process.env.API_KEY;
  const secretKey = process.env.API_SECRET;
  const ts = new Date().toISOString().split('.')[0] + 'Z';
  const nonce = Date.now().toString();
  const method = "/v3/orders/cancel";
  const apiUrl = "api.ox.fun";

  const postData = {
    recvWindow: 200000,
    responseType: "FULL",
    timestamp: Date.now(),
    orders: orders
  };

  const body = JSON.stringify(postData);
  const msgString = `${ts}\n${nonce}\nDELETE\n${apiUrl}\n${method}\n${body}`;

  const sign = crypto.createHmac('sha256', secretKey)
    .update(msgString)
    .digest('base64');

  const headers = {
    'Content-Type': 'application/json',
    'AccessKey': apiKey,
    'Timestamp': ts,
    'Signature': sign,
    'Nonce': nonce
  };

  try {
    const response = await axios.delete(`https://${apiUrl}${method}`, {
      data: body,
      headers: headers
    });
    console.log('Response:', response.data);
  } catch (error) {
    console.error('Error canceling orders:', error.response ? error.response.data : error.message);
  }
}

// Example usage
const ordersToCancel = [
  {
    marketCode: "BTC-USD-SWAP-LIN",
    orderId: "1111111111",
    clientOrderId: 11111111111111
  }
];

cancelOrders(ordersToCancel);